File: velocity-AUC-curve/velocity_AUC_visualize.py
import os
import json
import matplotlib.pyplot as plt
from numpy import mean

def visualize():
    with open(os.path.join(os.path.dirname(os.path.abspath(__file__)), 'velocity_AUC_curve.json'), 'r') as file:
        loaded_data = json.load(file)
        AUCcurveESTMD = loaded_data['AUCcurveESTMD']
        AUCcurveDSTMD = loaded_data['AUCcurveDSTMD']
        AUCcurveFracSTMD = loaded_data['AUCcurveFracSTMD']
        AUCcurveBackbonev2 = loaded_data['AUCcurveBackbonev2']

    V_LIST = range(50,2024,50)
    TAU_LIST = [1,5,11,19,29]

    fig1, ax1 = plt.subplots()
    fig2, ax2 = plt.subplots()
    fig3, ax3 = plt.subplots()
    fig4, ax4 = plt.subplots()

    for i in range(5):
        mAUC_ESTMD = mean(AUCcurveESTMD[i])
        ax1.plot(V_LIST, AUCcurveESTMD[i], label='tau=%d, mAUC=%0.2f'%(TAU_LIST[i], mAUC_ESTMD))
        mAUC_DSTMD = mean(AUCcurveDSTMD[i])
        ax2.plot(V_LIST, AUCcurveDSTMD[i], label='tau=%d, mAUC=%0.2f'%(TAU_LIST[i], mAUC_DSTMD))
        mAUC_FracSTMD = mean(AUCcurveFracSTMD[i])
        ax3.plot(V_LIST, AUCcurveFracSTMD[i], label='tau=%d, mAUC=%0.2f'%(TAU_LIST[i], mAUC_FracSTMD))
    mAUC_Backbonev2 = mean(AUCcurveBackbonev2)
    ax4.plot(V_LIST, AUCcurveBackbonev2 , 'r-*', linewidth=2, markeredgewidth=2, label='Proposed, mAUC=%0.2f'%mAUC_Backbonev2)

    ax1.set_xlim(0, 2000)
    ax1.set_ylim(0, 1)
    ax1.set_xlabel('velocity (pixels/s)')
    ax1.set_ylabel('AUC')
    ax1.title.set_text('ESTMD')
    ax1.legend()

    ax2.set_xlim(0, 2000)
    ax2.set_ylim(0, 1)
    ax2.set_xlabel('velocity (pixels/s)')
    ax2.set_ylabel('AUC')
    ax2.title.set_text('DSTMD')
    ax2.legend()

    ax3.set_xlim(0, 2000)
    ax3.set_ylim(0, 1)
    ax3.set_xlabel('velocity (pixels/s)')
    ax3.set_ylabel('AUC')
    ax3.title.set_text('FracSTMD')
    ax3.legend()

    ax4.set_xlim(0, 2000)
    ax4.set_ylim(0, 1)
    ax4.set_xlabel('velocity (pixels/s)')
    ax4.set_ylabel('AUC')
    ax4.title.set_text('Proposed')
    ax4.legend()
    
    plt.show()

if __name__ == '__main__':
    visualize()

File: velocity-AUC-curve/velocity_AUC_visualize.m
% 调用主函数
clc,clear,close all;

% 读取第一个 JSON 文件
data1 = jsondecode(fileread('velocity_AUC_curve.json'));
AUCcurveESTMD = data1.AUCcurveESTMD;
AUCcurveDSTMD = data1.AUCcurveDSTMD;
AUCcurveFracSTMD = data1.AUCcurveFracSTMD;
AUCcurveSTMDNet = data1.AUCcurveSTMDNet;

% 速度范围
vList = 50:50:2000;
TAU_LIST = [1,5,11,19,29];

% 创建子图
figure;
ax1 = subplot(2, 2, 1);
ax2 = subplot(2, 2, 2);
ax3 = subplot(2, 2, 3);
ax4 = subplot(2, 2, 4);

% 绘制曲线
for i = 1:5
    plot(ax1, vList, AUCcurveESTMD(i, :), 'DisplayName', sprintf('tau=%d, mAUC = %0.2f', TAU_LIST(i), mean(AUCcurveESTMD(i, :))));
    hold(ax1, 'on');
    plot(ax2, vList, AUCcurveDSTMD(i, :), 'DisplayName', sprintf('tau=%d, mAUC = %0.2f', TAU_LIST(i), mean(AUCcurveDSTMD(i, :))));
    hold(ax2, 'on');
    plot(ax3, vList, AUCcurveFracSTMD(i, :), 'DisplayName', sprintf('tau=%d, mAUC = %0.2f', TAU_LIST(i), mean(AUCcurveFracSTMD(i, :))));
    hold(ax3, 'on');
end
plot(ax4, vList, AUCcurveSTMDNet, 'r-*', 'LineWidth', 2, 'MarkerEdgeColor', 'r',...
    'DisplayName', sprintf('without tau, mAUC = %0.2f', mean(AUCcurveSTMDNet)));

% 设置图例和轴标签
set([ax1, ax2, ax3, ax4], 'XLim', [0 2000], 'YLim', [0 1]);
xlabel(ax1, 'velocity (pixels/s)');
ylabel(ax1, 'AUC');
title(ax1, 'ESTMD');
legend(ax1);

xlabel(ax2, 'velocity (pixels/s)');
ylabel(ax2, 'AUC');
title(ax2, 'DSTMD');
legend(ax2);

xlabel(ax3, 'velocity (pixels/s)');
ylabel(ax3, 'AUC');
title(ax3, 'FracSTMD');
legend(ax3, 'Location', 'SouthEast');

xlabel(ax4, 'velocity (pixels/s)');
ylabel(ax4, 'AUC');
title(ax4, 'Proposed');
legend(ax4, 'Location', 'SouthEast');

% 显示图像
hold off;



